packages/skeleton-react/src/lib/components/Rating/Rating.tsx

```diff
@@ -1,155 +1,72 @@
+		}),
+		{
+			context: zagProps
+		}
 	);
 
-	// https://www.w3.org/WAI/ARIA/apg/patterns/radio/examples/radio-rating/#kbd_label
-	const onRatingKeyDown = useCallback(
-		(event: React.KeyboardEvent<HTMLButtonElement>) => {
-			// If functions are not used outside of an effect/callback, they need to be defined in the effect/callback
-			function increaseValue() {
-				valueRef.current = Math.min(max, valueRef.current + 1 / step);
-				onValueChange(valueRef.current);
-				refreshFocus();
-			}
-
-			function decreaseValue() {
-				valueRef.current = Math.max(0, valueRef.current - 1 / step);
-				onValueChange(valueRef.current);
-				refreshFocus();
-			}
-
-			function refreshFocus() {
-				if (!figureRef.current) return;
-
-				const buttons = figureRef.current.querySelectorAll('button');
-				buttons[Math.max(0, Math.ceil(valueRef.current - 1))].focus();
-			}
-
-			if (!figureRef.current) return;
-			const rtl = getComputedStyle(figureRef.current).direction === 'rtl';
-			if (['ArrowLeft', 'ArrowUp'].includes(event.key)) {
-				event.preventDefault();
-				rtl ? increaseValue() : decreaseValue();
-			}
-			if (['ArrowRight', 'ArrowDown'].includes(event.key)) {
-				event.preventDefault();
-				rtl ? decreaseValue() : increaseValue();
-			}
-			onKeyDown(event);
-		},
-		[onKeyDown, max, onValueChange, step]
-	);
+	const api = rating.connect(state, send, normalizeProps);
 
 	return (
-		<figure ref={figureRef} className={`${base} ${width} ${justify} ${spaceX} ${classes}`} data-testid="rating">
-			{[...Array(max)].map((_, order) => (
-				<button
-					className={`${buttonBase} ${buttonPosition} ${buttonAspect} ${buttonClasses} ${interactive ? undefined : 'pointer-events-none'}`}
-					tabIndex={interactive && order === focusedButtonIndex ? 0 : -1}
-					onMouseDown={(event: React.MouseEvent<HTMLButtonElement>) => (interactive ? onRatingMouseDown(event, order) : undefined)}
-					onKeyDown={(event: React.KeyboardEvent<HTMLButtonElement>) => (interactive ? onRatingKeyDown(event) : undefined)}
-					type="button"
-					key={order}
-				>
-					<span
-						className={`${emptyBase} ${emptyClip} ${rxEmptyInteractive} ${emptyClasses}`}
-						style={
-							{
-								'--clipValue': `${(value - order) * 100}%`
-							} as React.CSSProperties
-						}
-						data-testid="rating-iconempty"
-					>
-						{iconEmpty}
-					</span>
-					<span
-						className={`${fullBase} ${fullClip} ${rxFullInteractive} ${fullClasses}`}
-						style={
-							{
-								'--clipValue': `${100 - (value - order) * 100}%`
-							} as React.CSSProperties
+		<div {...api.getRootProps()}>
+			{!!label && (
+				<label className={`${labelBase} ${labelClasses}`} {...api.getLabelProps()}>
+					{label}
+				</label>
+			)}
+			<div className={`${base} ${classes}`} {...api.getControlProps()}>
+				{api.items.map((index) => {
+					const itemState = api.getItemState({ index });
+
+					const icon = (() => {
+						if (!itemState.highlighted) {
```

Comment:
Consider a switch/case here if JSX allows it.

Reply:
JSX allows it but a switch isn't applicable here, since we're checking different types of states, not the same state that could have different values.